src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs

```diff
@@ -446,9 +446,16 @@ private T GetOptionValue<T>(string name)
             if (Options != null && Options.ContainsKey(name))
             {
                 object value = Options[name];
-                if(value != null && ( typeof(T) == typeof(int) || typeof(T) == typeof(int?)))
+                if (value != null && (typeof(T) != value.GetType()))
                 {
-                    value = System.Convert.ToInt32(value);
+                    if (typeof(T) == typeof(int) || typeof(T) == typeof(int?))
```

Comment:
I'm not sure how, convert.changetype takes the type as it's second parameter, so you should be able to specify `int` as the type to convert to.